### PR TITLE
feat: trace v2 clickhouse queries

### PR DESCRIPTION
**Pull request by nityanandagohain**
Part of https://github.com/SigNoz/signoz/pull/6537

* Add the note on how to add bucket filters.
* updated the traces queries
* removed old ch schema for logs and also added bucketing for logs

### CODE REVIEW

**Review by yuvraajsj18 — APPROVED** [review]

File: data/docs/userguide/writing-clickhouse-traces-query.mdx [hunk 1]
```diff
@@ -6,127 +6,142 @@ id: writing-clickhouse-traces-query
...
+|**attribute_string_rpc$$method** | RPC method of the span. Derived from `attributes_string['rpc.method']` attribute of a span|
+|**attribute_string_peer$$service** | Peer service of the span. Derived from `attributes_string['peer.service']` attribute of a span|
+
+## Timestamp Bucketing
+In the new schema for traces i.e `distributed_signoz_index_v3`, we have a new column 
+`ts_bucket_start` which is used to store the start timestamp of the bucket. This is used for faster filtering on timestamp attribute.
+
+How to use this column in the queries?
+If your timestamp query is `timestamp BETWEEN {{.start_datetime}} - 1800 AND {{.end_datetime}}` then you will have to add 
+ `ts_bucket_start BETWEEN {{.start_timestamp}} AND {{.end_timestamp}}`
```
Comment (srikanthccv) [inline data/docs/userguide/writing-clickhouse-traces-query.mdx]:
```suggestion
If your timestamp query is `timestamp BETWEEN {{.start_datetime}} AND {{.end_datetime}}` then you will have to add 
 `ts_bucket_start BETWEEN {{.start_timestamp}} - 1800 AND {{.end_timestamp}}`
```

File: data/docs/userguide/writing-clickhouse-traces-query.mdx [hunk 2]
```diff
@@ -299,9 +315,10 @@ The query is similar to timeseries query but instead of using time interval we u
...
 ```sql
 SELECT now() as ts, 
-stringTagMap['example_string_attribute'] AS attribute_name, 
-toFloat64(avg(durationNano)) AS value 
-FROM signoz_traces.distributed_signoz_index_v2  
+attributes_string['example_string_attribute'] AS attribute_name, 
+toFloat64(avg(duration_nano)) AS value 
+FROM signoz_traces.distributed_signoz_index_v3  
 WHERE timestamp BETWEEN {{.start_datetime}} AND {{.end_datetime}}
+AND ts_bucket_start BETWEEN {{.start_timestamp}} AND {{.end_timestamp}}
```
Comment (srikanthccv) [inline data/docs/userguide/writing-clickhouse-traces-query.mdx]: 1800 missing?

File: data/docs/userguide/writing-clickhouse-traces-query.mdx [hunk 3]
```diff
@@ -253,10 +268,11 @@ SELECT fromUnixTimestamp64Milli(intDiv( toUnixTimestamp64Milli ( timestamp ), 10
...
  SELECT timestamp 
- FROM signoz_traces.distributed_signoz_index_v2 
- WHERE serviceName='frontend' 
- AND durationNano>=50*exp10(6) 
- AND timestamp BETWEEN {{.start_datetime}} AND {{.end_datetime}})
+ FROM signoz_traces.distributed_signoz_index_v3
+ WHERE resource_string_service$$name='frontend' 
+ AND duration_nano>=50*exp10(6) 
+ AND timestamp BETWEEN {{.start_datetime}} AND {{.end_datetime}}
+ AND ts_bucket_start BETWEEN {{.start_timestamp}} AND {{.end_timestamp}})
```
Comment (srikanthccv) [inline data/docs/userguide/writing-clickhouse-traces-query.mdx]: Same

File: data/docs/userguide/writing-clickhouse-traces-query.mdx [hunk 4]
```diff
@@ -6,127 +6,142 @@ id: writing-clickhouse-traces-query
...
+    INDEX attributes_number_idx_val mapValues(attributes_number) TYPE bloom_filter GRANULARITY 1,
+    INDEX attributes_bool_idx_key mapKeys(attributes_bool) TYPE tokenbf_v1(1024, 2, 0) GRANULARITY 1,
+    INDEX resources_string_idx_key mapKeys(resources_string) TYPE tokenbf_v1(1024, 2, 0) GRANULARITY 1,
+    INDEX resources_string_idx_val mapValues(resources_string) TYPE ngrambf_v1(4, 5000, 2, 0) GRANULARITY 1
 )
+ENGINE = ReplicatedMergeTree('/clickhouse/tables/{uuid}/{shard}', '{replica}')
+PARTITION BY toDate(timestamp)
+ORDER BY (ts_bucket_start, resource_fingerprint, has_error, name, timestamp)
+TTL toDateTime(timestamp) + toIntervalSecond(1295000)
+SETTINGS index_granularity = 8192, ttl_only_drop_parts = 1
```
Comment (srikanthccv) [inline data/docs/userguide/writing-clickhouse-traces-query.mdx]:
```suggestion
ORDER BY (ts_bucket_start, resource_fingerprint, has_error, name, timestamp)
```

let's keep the relevant parts only

**Review by srikanthccv — APPROVED** [review]

**Review by Calm-Rock — APPROVED** [review]

**Pull request merged**